package/src/integration.ts

```diff
@@ -12,48 +27,135 @@ export default defineIntegration({
   optionsSchema,
   plugins: [...corePlugins],
   setup({ options }) {
-	const { resolve } = createResolver(import.meta.url)
+	// Create resolve helper
+	const { resolve } = createResolver(import.meta.url);
+
+	// Check if verbose logging is enabled
+	const isVerbose = options.verbose;
+	
+	// Create Gist Logger interface
+	const gistLogger = async (
+		logger: AstroIntegrationLogger, 
+		type: "info"|"warn"|"error", 
+		message: string,
+		checkVerbose: boolean,
+		) => {
+			// if checkVerbose is true and isVerbose is true, log the message
+			if (checkVerbose && isVerbose) {
```

Comment:
Hmmm I'm not sold on this solution, but this could just be one check with

`if (!checkVerbose || (checkVerbose && isVerbose)) {`

I'm not mega sold, but basically could refactor to remove the duplicate logging code